dot/network/notifications.go

```diff
@@ -152,9 +152,12 @@ func createDecoder(info *notificationsProtocol, handshakeDecoder HandshakeDecode
 }
 
 // createNotificationsMessageHandler returns a function that is called by the handler of *inbound* streams.
-func (s *Service) createNotificationsMessageHandler(info *notificationsProtocol,
+func (s *Service) createNotificationsMessageHandler(
+	info *notificationsProtocol,
 	messageHandler NotificationsMessageHandler,
-	batchHandler NotificationsMessageBatchHandler) messageHandler {
+	batchHandler NotificationsMessageBatchHandler,
+) messageHandler {
+
 	return func(stream libp2pnetwork.Stream, m Message) error {
 		if m == nil || info == nil || info.handshakeValidator == nil || messageHandler == nil {
```

Comment:
is it possible to add a defer to close this stream?

```go
return func(stream libp2pnetwork.Stream, m Message) error {
  defer func(){
    if err := stream.CloseWrite(); err != nil {
      logger.Tracef("failed to close stream for writing: %s", err)
    }
  }()
}
```

Reply:
Do we want to close the stream in every return case :thinking: ? How about `CloseRead`?

Reply:
hmm, got it!

Reply:
@EclesioMeloJunior how about `CloseRead`?

Reply:
@qdm12 @kishansagathiya at line 354 from `dot/network/notifications.go` we open a stream, we could add a `defer` which will call the `CloseRead` when the function exits

Reply:
`messageHandler` returned by `createNotificationsMessageHandler` is used in `readStream`.

`readStream` always resets the stream in the end using defer. When you reset the stream, the stream gets closed from both ends.

https://github.com/ChainSafe/gossamer/blob/f091a756d24663000b4ca99ba39e4aafbde95e84/dot/network/inbound.go#L14

Please unresolve, if you feel this still needs to be addressed.